src/scorer.py

```diff
@@ -58,10 +58,33 @@ def score(self, cand_o: Optional[str]) -> Tuple[int, str]:
         deletions = []
         insertions = []
 
+        def lo_hi_match(old: str, new:str):
+            old_lo = old.find("%lo")
+            old_hi = old.find("%hi")
+            new_lo = new.find("%lo")
+            new_hi = new.find("%hi")
+
+            if (old_lo != -1 and new_lo != -1):
+                old_idx = old_lo
+                new_idx = new_lo
+            elif (old_hi != -1 and new_hi != -1):
+                old_idx = old_hi
+                new_idx = new_hi
+            else:
+                return False
+
+            old_inner = old[old_idx + 4 : -1]
+            new_inner = new[new_idx + 4 : -1]
+            return old_inner == ".rodata" or new_inner == ".rodata"
```

Comment:
We should also allow `%lo(.rodata + 0x10)` and `%lo(.data + ...)` -- how about `old_inner.startswith(".") or new_inner.startswith(".")`?

Reply:
Compare the parts before old_idx and new_idx as well, they may contain regalloc diffs

Reply:
as for the parts before old_idx and new_idx, what should the behavior be if they're the same vs different?

Reply:
I think it would be fine to just return false here if they are different, thus resulting in PENALTY_REGALLOC if they differ (and else score 0).